src/tool-classes.ts

```diff
@@ -0,0 +1,179 @@
+import * as semver from 'semver';
+import * as core from '@actions/core';
+import dayjs from 'dayjs';
+import fetch from 'node-fetch';
+import { dateGte, isDateMoreThanSixMonthsAway } from "./utils";
+import { GitHubIssue, InternalRepository, ManifestRepository } from './repository-classes';
+
+interface IResponseFormat {
+    eol: string;
+    latest: string;
+    latestReleaseDate: string;
+    lts: boolean | string;
+}
+
+abstract class Tool {
+    constructor(
+        protected name: string, 
+        protected eolApiEndpoint: string, 
+        protected manifestRepository: ManifestRepository,
+        protected internalRepository: InternalRepository = new InternalRepository()
+    ) {
+        this.name = name;
+        this.eolApiEndpoint = eolApiEndpoint;
+        this.manifestRepository = manifestRepository;
+    }
+
+    protected async getVersionsFromApi(url: string): Promise<IResponseFormat[]> {
+        const response = await fetch(url);
+        const data: IResponseFormat[] = await response.json() as IResponseFormat[];
+        return data;
+    }
+
+    protected async filterApiData(data: IResponseFormat[]) {
+        const filteredData = data.filter((item: IResponseFormat) => {
+            const eolDate = new Date(item.eol);
+            // The condition below is needed as 'lts: false' for Node means that the version is unstable (e.g. v15)
+            // while in the response for Python and Go, all versions have 'lts' set to false and it would return undefined.
+            const condition = this.name === 'Node' ? 
+                (dateGte(eolDate, new Date()) && item.lts !== false) : 
+                dateGte(eolDate, new Date());
+            return condition;
+        }).reverse();
+        return filteredData;
+    }
+
+    async checkVersions() {
+        const toolVersionsFromEolApi = await this.getVersionsFromApi(this.eolApiEndpoint);
+        const filteredToolVersionsFromEolApi = await this.filterApiData(toolVersionsFromEolApi);
+        const versionClosestToEol = filteredToolVersionsFromEolApi[0];
+        
+        core.info(`\n ${this.name} version: ${versionClosestToEol.latest}`);
+        core.info(` For more info on ${this.name} versions, please visit: https://endoflife.date/${this.name === 'Node' ? 'nodejs' : 'python' }\n`);
+        
+        const manifestData = await this.manifestRepository.getVersionsManifestFromRepo(versionClosestToEol.latest);
+        const earliestVersionInManifest = manifestData[0].version;
+    
+        if (!semver.gte(versionClosestToEol.latest, earliestVersionInManifest)) {
+            core.info(`The version of ${this.name} (${versionClosestToEol.latest}) provided by the API does not match the one in the manifest (${earliestVersionInManifest}).\n`);
+            core.warning(`The version of ${this.name} provided by the API is ${versionClosestToEol.latest} and the one in the manifest is ${earliestVersionInManifest}.`);
```

Comment:
It seems to me that these logs are about the same, so I would get rid of one of them
```suggestion
            core.warning(`The version of ${this.name} (${versionClosestToEol.latest}) provided by the API does not match the one in the manifest (${earliestVersionInManifest}).\n`);
```

Reply:
Fixed